static/js/components/material/Dialog.js

```diff
@@ -22,36 +22,47 @@ export default class Dialog extends React.Component {
   // $FlowFixMe: Flow doesn't like the extra props that aren't part of mdc.dialog class
   props: DialogProps;
 
-  componentDidMount() {
-    const { open, onAccept, onCancel } = this.props;
-
+  createDialog() {
+    const { onAccept, onCancel } = this.props;
     // Hack to get dialog to play nicely with JS tests
     if (!this.dialogRoot || !this.dialogRoot.dataset) return;
-
     this.dialog = new MDCDialog(this.dialogRoot);
     this.dialog.listen('MDCDialog:accept', onAccept);
     // $FlowFixMe: Flow thinks this.dialog might be null
     this.dialog.listen('MDCDialog:cancel', onCancel);
-    if (open) {
-      // $FlowFixMe: Flow thinks this.dialog might be null
+  }
+
+  showMdc() {
+    if (this.dialog) {
       this.dialog.show();
     }
   }
 
-  componentWillUnmount() {
+  destroyMdc() {
     if (this.dialog) {
       this.dialog.destroy();
     }
   }
 
+  componentDidMount() {
+    const { open } = this.props;
+    this.createDialog();
+    if (open) {
+      this.showMdc();
+    }
+  }
+
+  componentWillUnmount() {
+    this.destroyMdc();
+  }
+
   componentWillReceiveProps(nextProps: DialogProps) {
-    if (this.dialog) {
-      if (this.props.open !== nextProps.open) {
-        if (nextProps.open) {
-          this.dialog.show();
-        } else {
-          this.dialog.close();
-        }
+    if (this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.createDialog();
```

Comment:
I tested this without `this.createDialog()` and it seemed to work. Are you sure we need this?

Reply:
Guess not, I could have sworn that I wasn't able to reopen dialogs without it earlier today, but now it does seem to be working fine without it.